features/futures/pci-passthrough.md

```diff
@@ -0,0 +1,94 @@
+
+12:14
+
+
+
+
+
+
+
+---
+title: PCI passthrough support
+layout: default
+design_doc: true
+revision: 1
+status: proposed
+---
+
+Introduction
+------------
+
+GPU passthrough is already available in XAPI, this document proposes to also
+offer passthrough for all PCI devices through XAPI.
+
+Design proposal
+---------------
+
+New methods for PCI object:
+- `PCI.enable_dom0_access`
+- `PCI.disable_dom0_access`
+- `PCI.get_dom0_access_status`: compares the outputs of `/opt/xensource/libexec/xen-cmdline`
```

Comment:
Note that the name diverges from the one in `PGPU`; I have no strong opinions; `get_dom0_access` could be enough.

Reply:
The document details the API but not the CLI to support this. The API should be agreed on first.

Reply:
The idea is to lose the DB field to be able to reflect the state of the accss even if manually modified without using XAPI by a user.
We can keep the `get_dom0_access` if preferred but we wantd to mark the difference with the db field.